src/Main.cpp

```diff
@@ -23,14 +30,193 @@
  |                                      + 3 : Troposcatter
  |
  *===========================================================================*/
-int Main(double d__km, double h_1__meter, double h_2__meter, double f__mhz, double time_percentage, Result* result)
+int P528(double d__km, double h_1__meter, double h_2__meter, double f__mhz, double time_percentage, Result *result)
 {
-    Terminal terminal_1;
-    Terminal terminal_2;
-    TroposcatterParams tropo;
-    Path path;
-    LineOfSightParams los_params;
-
-    return MainEx(d__km, h_1__meter, h_2__meter, f__mhz, time_percentage, result,
-        &terminal_1, &terminal_2, &tropo, &path, &los_params);
+	Terminal terminal_1;
+	Terminal terminal_2;
+	TroposcatterParams tropo;
+	Path path;
+	LineOfSightParams los_params;
+
+	// reset Results struct
+	result->A_fs__db = 0;
+	result->A__db = 0;
+	result->d__km = 0;
+	result->los_iterations = 0;
+	result->propagation_mode = PROP_MODE__NOT_SET;
+
+	int err = ValidateInputs(d__km, h_1__meter, h_2__meter, f__mhz, time_percentage);
+    if (err != SUCCESS)
+    {
+        if (err == ERROR_HEIGHT_AND_DISTANCE)
+        {
+            result->A_fs__db = 0;
+            result->A__db = 0;
+            result->d__km = 0;
+            return SUCCESS;
+        }
+        else
+            return err;
+    }
+
+	double N_s = N_0;
+	path.a_e__km = a_0__km / (1.0 - 0.04665 * exp(0.005577 * N_s));
```

Comment:
Magic numbers. It would be helpful to the reader to have a comment explain them and/or their source.